test/unit/math/prim/prob/inv_wishart_cholesky_rng_test.cpp

```diff
@@ -0,0 +1,129 @@
+#include <stan/math/prim.hpp>
+#include <test/unit/math/prim/util.hpp>
+#include <boost/random/mersenne_twister.hpp>
+#include <boost/math/distributions/chi_squared.hpp>
+#include <gtest/gtest.h>
+#include <stdexcept>
+#include <vector>
+
+TEST(ProbDistributionsInvWishartCholesky, rng) {
+  using Eigen::Dynamic;
+  using Eigen::Matrix;
+  using Eigen::MatrixXd;
+
+  using stan::math::inv_wishart_cholesky_rng;
+
+  boost::random::mt19937 rng;
+
+  MatrixXd omega(3, 4);
+  EXPECT_THROW(inv_wishart_cholesky_rng(3.0, omega, rng), std::domain_error);
+
+  MatrixXd sigma(3, 3);
+  sigma << 9.0, -3.0, 0.0, -3.0, 4.0, 1.0, 0.0, 1.0, 3.0;
+
+  Matrix<double, Dynamic, Dynamic> LS = sigma.llt().matrixL();
+
+  EXPECT_NO_THROW(inv_wishart_cholesky_rng(3.0, LS, rng));
+  EXPECT_THROW(inv_wishart_cholesky_rng(2, LS, rng), std::domain_error);
+  EXPECT_THROW(inv_wishart_cholesky_rng(-1, LS, rng), std::domain_error);
+  LS(2, 2) = -1.0;
+  EXPECT_THROW(inv_wishart_cholesky_rng(3.0, LS, rng), std::domain_error);
+}
+
+TEST(ProbDistributionsInvWishartCholesky, rng_pos_def) {
+  using Eigen::MatrixXd;
+  using stan::math::inv_wishart_cholesky_rng;
+
+  boost::random::mt19937 rng;
+
+  MatrixXd Sigma(2, 2);
+  MatrixXd Sigma_non_pos_def(2, 2);
+
+  Sigma << 1, 0, 0, 1;
+  Sigma_non_pos_def << -1, 0, 0, 1;
+
+  unsigned int dof = 5;
+
+  EXPECT_NO_THROW(inv_wishart_cholesky_rng(dof, Sigma, rng));
+  EXPECT_THROW(inv_wishart_cholesky_rng(dof, Sigma_non_pos_def, rng),
+               std::domain_error);
+}
+
+TEST(ProbDistributionsInvWishartCholesky, cholesky_factor_check) {
+  using Eigen::MatrixXd;
+  using stan::math::identity_matrix;
+  using stan::math::inv_wishart_cholesky_rng;
+  using stan::test::unit::expect_symmetric;
+  using stan::test::unit::spd_rng;
+
+  boost::random::mt19937 rng;
+
+  for (int k = 1; k < 20; ++k)
+    for (double nu = k - 0.9; nu < k + 10; ++nu)
+      for (int n = 0; n < 10; ++n)
+        expect_symmetric(stan::math::multiply_lower_tri_self_transpose(
```

Comment:
Isn't this guaranteed to hold because it's the product of a matrix and its transpose?  Or to ask this another way, what can go wrong here?